state/protocol/badger/snapshot.go

```diff
@@ -295,12 +295,48 @@ func (s *Snapshot) Pending() ([]flow.Identifier, error) {
 	return s.pending(s.blockID)
 }
 
-func (s *Snapshot) pending(blockID flow.Identifier) ([]flow.Identifier, error) {
+func (s *Snapshot) ValidPending() ([]flow.Identifier, error) {
+	pendingIDs, err := s.lookupChildren(s.blockID)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.validPending(pendingIDs)
+}
+
+func (s *Snapshot) lookupChildren(blockID flow.Identifier) ([]flow.Identifier, error) {
 	var pendingIDs []flow.Identifier
 	err := s.state.db.View(procedure.LookupBlockChildren(blockID, &pendingIDs))
 	if err != nil {
 		return nil, fmt.Errorf("could not get pending children: %w", err)
 	}
+	return pendingIDs, nil
+}
+
+func (s *Snapshot) validPending(pendingIDs []flow.Identifier) ([]flow.Identifier, error) {
+	var validPendingIds []flow.Identifier
+	for _, pendingID := range pendingIDs {
+		grandchildren, err := s.lookupChildren(pendingID)
```

Comment:
is the reason of calling `grandchildren` is that all pendingIDs are from the same node, and the grandchildren here are relative to that common parent of all pendingIDs?